diff --git a/mod/forum/lib.php b/mod/forum/lib.php
index bc54e22..9efab6c 100644
--- a/mod/forum/lib.php
+++ b/mod/forum/lib.php
@@ -3294,6 +3294,12 @@ function forum_print_post($post, $discussion, $forum, &$cm, $course, $ownpost=fa
     // Prepare an array of commands
     $commands = array();
 
+    // BEGIN moodle.org hack MDLSITE-1620
+    if ($reply) {
+        $commands[] = html_writer::link(new moodle_url('/blocks/spam_deletion/reportspam.php', array('postid' => $post->id)), get_string('reportasspam', 'block_spam_deletion'));
+    }
+    // END moodle.org hack MDLSITE-1620
+
     // SPECIAL CASE: The front page can display a news item post to non-logged in users.
     // Don't display the mark read / unread controls in this case.
     if ($istracked && $CFG->forum_usermarksread && isloggedin()) {
